src/pkg/clouds/aws/ecs/logs.go

```diff
@@ -293,41 +293,53 @@ type EventStream interface {
 }
 
 type collectionStream struct {
-	streams []EventStream
+	cancel  context.CancelFunc
 	ch      chan types.StartLiveTailResponseStream
+	ctx     context.Context // derived from the context passed to TailLogGroups
 	errCh   chan error
+	streams []EventStream
 
-	done chan struct{}
 	lock sync.Mutex
 	wg   sync.WaitGroup
 }
 
-func (c *collectionStream) addAndStart(s EventStream) {
+func (c *collectionStream) addAndStart(s EventStream, since time.Time, lgi LogGroupInput) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 	c.streams = append(c.streams, s)
 	c.wg.Add(1)
 	go func() {
 		defer c.wg.Done()
+		if !since.IsZero() {
+			// Query the logs between the start time and now
+			if events, err := Query(c.ctx, lgi, since, time.Now()); err != nil {
```

Comment:
I think we can add an `nextToken` param to Query function and put this in a loop.

Reply:
agree; already have a TODO in there